src/sync/sync_config.hpp

```diff
@@ -100,6 +104,7 @@ struct SyncConfig {
     SyncSessionStopPolicy stop_policy;
     std::function<SyncBindSessionHandler> bind_session_handler;
     std::function<SyncSessionErrorHandler> error_handler;
+    std::shared_ptr<ChangesetTransformer> transformer;
```

Comment:
Alright, I will make the change you suggest. I'll say though, that I dislike the change on principal grounds, because it obscures the point of destruction, and forces you to dynamically allocate the cooker. I understand that this is no problem for the current use case. I also understand that there is little you can do on your side to solve the problem given the current limitations of the `sync::Session` API. I hope in the future we can revisit this and find a better solution.